src/azure/client.go

```diff
@@ -167,7 +167,7 @@ func (az *azClient) buildFirewallConfig(erulesList egressv1.EgressrulesList, eru
 			} else {
 				for i := 0; i < len(ruleCollections); i++ {
 					ruleCollection := ruleCollections[i].(*n.FirewallPolicyFilterRuleCollection)
-					if az.getRuleCollectionType(erule.Spec.RuleType) == ruleCollection.RuleCollectionType && *az.buildAction(erule.Spec.Action) == *ruleCollection.Action && strings.Contains(*ruleCollection.Name, erule.Spec.RuleType) {
+					if erule.Spec.RuleCollectionName == *ruleCollection.Name {
```

Comment:
Is this to check the presence of rule collection in a single rule collection group? Rule collection name is unique per RCG and different RCGs can have a rule collection with the same name.

Reply:
Okay Sujith. But for now since we are just supporting a single resource collection group shouldn't this be okay now? In the future when we want to support multiple RCG's we can have an additional condition to check for the presence of rule collection in a single rule collection group.

Reply:
Sounds good. Just wanted to call this out.